test/document.populate.test.js

```diff
@@ -934,4 +934,118 @@ describe('document.populate', function() {
     assert.ok(foundBook.populated('authorId'));
     assert.ok(foundBook.authorId.populated('websiteId'));
   });
+
+  it('works when populating a nested document inside an array parent (gh-14435)', async function() {
+    const CodeSchema = new Schema({
+      code: String
+    });
+
+    const UserSchema = new Schema({
+      username: String,
+      extras: [
+        new Schema({
+          config: new Schema({
+            paymentConfiguration: {
+              paymentMethods: [
+                {
+                  type: Schema.Types.ObjectId,
+                  ref: 'Code'
+                }
+              ]
+            }
+          })
+        })
+      ]
+    });
+
+    const Code = db.model('Code', CodeSchema);
+    const CodeUser = db.model('CodeUser', UserSchema);
+
+    const code = await new Code({
+      code: 'test code'
+    }).save();
+
+    await new CodeUser({
+      username: 'TestUser',
+      extras: [
+        {
+          config: {
+            paymentConfiguration: {
+              paymentMethods: [code._id]
+            }
+          }
+        }
+      ]
+    }).save();
```

Comment:
```suggestion
    await CodeUser.create({
      username: 'TestUser',
      extras: [
        {
          config: {
            paymentConfiguration: {
              paymentMethods: [code._id]
            }
          }
        }
      ]
    });
```
should likely be a `.create`, unless this syntax is explicitly meant to be tested?